app/models/order_detail.rb

```diff
@@ -752,7 +762,9 @@ def cancel_reservation(canceled_by, canceled_reason: nil, order_status: OrderSta
         change_status! order_status
       end
     else
+      return false if state.to_sym == :unreconciled
```

Comment:
Is this intended to be `:unreconciled`? 
I wonder if this is part of what is making specs fail
```suggestion
      return false if state.to_sym == :unrecoverable
```